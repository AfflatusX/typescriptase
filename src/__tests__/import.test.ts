import {
  importAll1,
  importAll2,
  importAll3,
  importDefault1,
  importDefault2,
  importDefault3,
  importRaw1,
  importRaw2,
  importRaw3,
  importSome1,
  importSome2,
  importSome3,
} from "./examples";

test(
  "importAll1",
  async (): Promise<void> => {
    expect(importAll1.print({
      name: "importAll1",
      path: "src/__tests__/import.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "importAll2",
  async (): Promise<void> => {
    expect(importAll2.print({
      name: "importAll2",
      path: "src/__tests__/import.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "importAll3",
  async (): Promise<void> => {
    expect(importAll3.print({
      name: "importAll3",
      path: "src/__tests__/import.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "importDefault1",
  async (): Promise<void> => {
    expect(importDefault1.print({
      name: "importDefault1",
      path: "src/__tests__/import.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "importDefault2",
  async (): Promise<void> => {
    expect(importDefault2.print({
      name: "importDefault2",
      path: "src/__tests__/import.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "importDefault3",
  async (): Promise<void> => {
    expect(importDefault3.print({
      name: "importDefault3",
      path: "src/__tests__/import.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "importRaw1",
  async (): Promise<void> => {
    expect(importRaw1.print({
      name: "importRaw1",
      path: "src/__tests__/import.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "importRaw2",
  async (): Promise<void> => {
    expect(importRaw2.print({
      name: "importRaw2",
      path: "src/__tests__/import.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "importRaw3",
  async (): Promise<void> => {
    expect(importRaw3.print({
      name: "importRaw3",
      path: "src/__tests__/import.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "importSome1",
  async (): Promise<void> => {
    expect(importSome1.print({
      name: "importSome1",
      path: "src/__tests__/import.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "importSome2",
  async (): Promise<void> => {
    expect(importSome2.print({
      name: "importSome2",
      path: "src/__tests__/import.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "importSome3",
  async (): Promise<void> => {
    expect(importSome3.print({
      name: "importSome3",
      path: "src/__tests__/import.test.ts",
    }))
      .toMatchSnapshot();
  },
);
