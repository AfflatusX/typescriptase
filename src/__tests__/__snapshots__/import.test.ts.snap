// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`importAll1 1`] = `"import * as zzz from \\"YYY\\";"`;

exports[`importAll2 1`] = `"import * as aaa from \\"BbB\\";"`;

exports[`importAll3 1`] = `"import * as mmm from \\"./mmm\\";"`;

exports[`importDefault1 1`] = `"import xxx from \\"www\\";"`;

exports[`importDefault2 1`] = `"import fff from \\"./ddd\\";"`;

exports[`importDefault3 1`] = `"import ooo from \\"nnn\\";"`;

exports[`importRaw1 1`] = `"import \\"ZZZ\\";"`;

exports[`importRaw2 1`] = `"import \\"./p\\";"`;

exports[`importRaw3 1`] = `"import \\"aqe\\";"`;

exports[`importSome1 1`] = `
"import {
  B,
  c,
} from \\"a\\";"
`;

exports[`importSome2 1`] = `
"import {
  A,
  wA,
} from \\"./R\\";"
`;

exports[`importSome3 1`] = `
"import {
  g,
  r,
  z,
} from \\"./a\\";"
`;
